src/Common/tests/System/Net/Sockets/SocketTestServerAsync.cs

```diff
@@ -188,8 +191,7 @@ private void ProcessAccept(SocketAsyncEventArgs e)
 
                 if (Interlocked.Decrement(ref _acceptRetryCount) <= 0)
                 {
-                    Assert.True(false, "accept retry limit exceeded.");
-                    return;
+                    throw new Exception("accept retry limit exceeded.");
```

Comment:
Sure, that's probably better.
I went looking through list of common exceptions, but wasn't able to find anything at the time that struck me as specifically appropriate otherwise.

I feel like what it wants is some sort of sibling exception to `TimeoutException`, maybe something to propose as a new API?